apps/api/src/database/repositories/admin-repository.ts

```diff
@@ -1,303 +1,10 @@
-    }
-
-    return result;
-  } catch (error) {
-    repositoryLogger.error("Error in setApiKey:", error);
-    throw error;
-  }
-}
-
-/**
- * Update last login timestamp for an admin
- * @param id Admin ID
- */
-async function updateLastLoginImpl(id: string): Promise<SelectAdmin> {
-  try {
-    const now = new Date();
-    const [result] = await db
-      .update(admins)
-      .set({
-        lastLoginAt: now,
-        updatedAt: now,
-      })
-      .where(eq(admins.id, id))
-      .returning();
-
-    if (!result) {
-      throw new Error("Failed to update last login - no result returned");
-    }
-
-    return result;
-  } catch (error) {
-    repositoryLogger.error("Error in updateLastLogin:", error);
-    throw error;
-  }
-}
-
-/**
- * Update password hash for an admin
- * @param id Admin ID
- * @param passwordHash New password hash
- */
-async function updatePasswordImpl(
-  id: string,
-  passwordHash: string,
-): Promise<SelectAdmin> {
-  try {
-    const now = new Date();
-    const [result] = await db
-      .update(admins)
-      .set({
-        passwordHash,
-        updatedAt: now,
-      })
-      .where(eq(admins.id, id))
-      .returning();
-
-    if (!result) {
-      throw new Error("Failed to update password - no result returned");
-    }
-
-    return result;
-  } catch (error) {
-    repositoryLogger.error("Error in updatePassword:", error);
-    throw error;
-  }
-}
-
-/**
- * Search for admins by various attributes
- * @param searchParams Object containing search parameters
- * @returns Array of admins matching the search criteria
- */
-async function searchAdminsImpl(
-  searchParams: SearchAdminsParams,
-): Promise<SelectAdmin[]> {
-  try {
-    const conditions = [];
-
-    // Add filters for each provided parameter
-    if (searchParams.username) {
-      conditions.push(ilike(admins.username, `%${searchParams.username}%`));
-    }
-
-    if (searchParams.email) {
-      conditions.push(ilike(admins.email, `%${searchParams.email}%`));
-    }
-
-    if (searchParams.name) {
-      conditions.push(ilike(admins.name, `%${searchParams.name}%`));
-    }
-
-    if (searchParams.status) {
-      conditions.push(eq(admins.status, searchParams.status));
-    }
-
-    // If no search parameters were provided, return all admins
-    if (conditions.length === 0) {
-      return await db.select().from(admins);
-    }
-
-    // Combine all conditions with AND operator
-    return await db
-      .select()
-      .from(admins)
-      .where(and(...conditions));
-  } catch (error) {
-    repositoryLogger.error("Error in searchAdmins:", error);
-    throw error;
-  }
-}
-
-/**
- * Count all admins
- */
-async function countImpl(): Promise<number> {
-  try {
-    const [result] = await db.select({ count: drizzleCount() }).from(admins);
-    return result?.count ?? 0;
-  } catch (error) {
-    repositoryLogger.error("Error in count:", error);
-    throw error;
-  }
-}
+const repository = new AdminRepository(db, repositoryLogger);
```

Comment:
The thing to notice about all the "repository" files that still live in the api app is that they are simply timing wrappers around the actual repositories that have been moved to the shared db package. Here we create the repository instance, passing in its dependencies.

Reply:
probably at some point we should have a RepositoryRegistry similar to the ServiceRegistry and construct all the repository instance in one place at startup rather than staticly within the files like this...
But this is good enough for now

Reply:
Yea I agree, but my goal here was to move the repositories to a shared package while minimizing how existing code uses them.